share/mk/bsd.lib.mk

```diff
@@ -153,17 +153,15 @@ PIEFLAG=-fpie
 PO_FLAG=-pg
 
 .c.po:
-	${CC.${.IMPSRC:T}:U${CC}} ${PO_FLAG} ${STATIC_CFLAGS} ${PO_CFLAGS} \
-	    -c ${.IMPSRC} -o ${.TARGET}
+	${CC} ${PO_FLAG} ${STATIC_CFLAGS} ${PO_CFLAGS} -c ${.IMPSRC} -o ${.TARGET}
```

Comment:
I'm pretty sure these were from when we compiled a few things in libc with CHERI clang, but most with the base compiler.